### PR TITLE
Add a make cluster-operator-roles target.

### PR DESCRIPTION
**Pull request by dgoodwin**
This target assumes we have uploaded the template to our namespace.
(which leaves it's contents in our control)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -209,3 +209,6 @@ image-pruner-setup:
 	oc adm policy --as=system:admin add-cluster-role-to-user system:image-pruner -z image-pruner
 	$(MAKE) apply WHAT=cluster/ci/jobs/image-pruner.yaml
 .PHONY: image-pruner-setup
+
+cluster-operator-roles:
```
Comment (smarterclayton) [inline Makefile]: Where is your role?

Comment (smarterclayton) [reply Makefile]: Your role and the definition belongs in this repo if you are asking for permissions.

Comment (dgoodwin) [reply Makefile]: The role lives with the app and all the ansible we use to deploy it: https://github.com/openshift/cluster-operator/blob/master/contrib/examples/cluster-operator-roles-template.yaml

My thought was it's pretty easy for us to upload the template with our ansible, as opposed to maintaining two copies of the same data in different repos. 

Is that however required for review purposes?

**Comment by dgoodwin** [comment]
Do you mean the deployment objects for our controllers? We have those yes but my intent was that my team manages that hands on using openshift itself. We have accounts on the cluster, but not cluster-admin privs, so we cannot create these objects. We broke them out into a separate *hopefully* relatively static template that we need to ask someone to run for us. Then we take over with deploying and maintaining the cluster operator itself. Once these are in play we can run our playbook to deploy and keep up to date.

**Comment by dgoodwin** [comment]
We built such that clusters can live in any namespace, or multiple per namespace, and our expectation was that we'd be doing a namespace per pull request here as that will really help keep track of jobs and clusters and their data, and clean it up, and debug when something went wrong. Each cluster needs several secrets (certs, creds, ssh key), and thus to support clusters in multiple namespaces we added the priv to read secrets in all namespaces. Thoughts?

**Comment by eparis** [comment]
I think the final answer is that we will need to run 1 CO API server. We would run 1 CO controller per namespace which is managing clusters. We'll need to write the cluster operator operator which would watch for users which create a cluster or clusterversion object ref in their namespace and automatically launch/manage a controller in their namespace. (or maybe we make them go to olm and manually install their own cluster operator controller operator, which installs the cluster operator controller)

In either case, I'm happy to get this redesign on our backlog, but it isn't going to happen in the timeframe we need to be running as part of the CI system.

**Comment by smarterclayton** [comment]
That's fine, I just need to know.  The ALM pattern is relevant to you, please talk to Jessica and about the tradeoffs and the goals.

File: Makefile [hunk 2]
```diff
@@ -10,7 +10,7 @@ applyTemplate:
 	oc process -f $(WHAT) | oc apply -f -
 .PHONY: applyTemplate
 
-all: roles prow projects
+all: roles prow projects cluster-operator-roles
```
Comment (smarterclayton) [inline Makefile]: make your dependency a sub dependency of roles

**Comment by smarterclayton** [comment]
One more comment

**Comment by dgoodwin** [comment]
Done!

**Comment by smarterclayton** [comment]
Permissions are created

File: projects/cluster-operator/cluster-operator-roles-template.yaml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    resourceNames: ["cluster-operator-controller-manager"]
+    verbs: ["get","update"]
+  # events for recording events
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs:     ["create","patch","update"]
+  # allow all operations on all resources in our API group
+  - apiGroups: ["clusteroperator.openshift.io"]
+    resources: ["*"]
+    verbs:     ["create", "get", "list", "watch", "update", "patch", "delete"]
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply projects/cluster-operator/cluster-operator-roles-template.yaml]: Probably not, we will adjust upstream and copy it over next time we sync.

**Comment by dgoodwin** [comment]
Thanks @smarterclayton !

**Pull request merged**